python/ray/serve/_private/controller.py

```diff
@@ -115,16 +117,30 @@ async def __init__(
         controller_name: str,
         *,
         http_config: HTTPOptions,
+        logging_config: LoggingConfig,
         grpc_options: Optional[gRPCOptions] = None,
     ):
         self._controller_node_id = ray.get_runtime_context().get_node_id()
         assert (
             self._controller_node_id == get_head_node_id()
         ), "Controller must be on the head node."
 
-        configure_component_logger(
-            component_name="controller", component_id=str(os.getpid())
-        )
+        self.ray_worker_namespace = ray.get_runtime_context().namespace
+        self.controller_name = controller_name
+        self.gcs_client = GcsClient(address=ray.get_runtime_context().gcs_address)
+        kv_store_namespace = f"{self.controller_name}-{self.ray_worker_namespace}"
+        self.kv_store = RayInternalKVStore(kv_store_namespace, self.gcs_client)
+
+        self.long_poll_host = LongPollHost()
+        self.done_recovering_event = asyncio.Event()
+
+        # Try to read config from checkpoint
+        self.logging_config = None
+        log_config_checkpoint = self.kv_store.get(LOGGING_CONFIG_CHECKPOINT_KEY)
+        if log_config_checkpoint is not None:
+            logging_config = pickle.loads(log_config_checkpoint)
+        self.reconfigure_system_logging_config(logging_config)
```

Comment:
please add a comment about the precedence and why it's structured this way (i.e., that the one in the constructor is ignored if `reconfigure` is ever called)